[Bug]: With a new Gradle plugins

### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-geolocator/blob/main/geolocator/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [ ] iOS
- [ ] Linux
- [ ] macOS
- [ ] Web
- [ ] Windows

### Steps to reproduce

I can't export APK

### Expected results

I created a new project and added this package to test it, but I cannot extract the app version.

### Actual results

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkReleaseDuplicateClasses'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable
   > Duplicate class kotlin.collections.jdk8.CollectionsJDK8Kt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.internal.jdk7.JDK7PlatformImplementations found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk7-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10)
     Duplicate class kotlin.internal.jdk7.JDK7PlatformImplementations$ReflectSdkVersion found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk7-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10)
     Duplicate class kotlin.internal.jdk8.JDK8PlatformImplementations found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.internal.jdk8.JDK8PlatformImplementations$ReflectSdkVersion found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.io.path.ExperimentalPathApi found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk7-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10)
     Duplicate class kotlin.io.path.PathRelativizer found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk7-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10)
     Duplicate class kotlin.io.path.PathsKt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk7-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10)
     Duplicate class kotlin.io.path.PathsKt__PathReadWriteKt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk7-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10)
     Duplicate class kotlin.io.path.PathsKt__PathUtilsKt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk7-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10)
     Duplicate class kotlin.jdk7.AutoCloseableKt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk7-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10)
     Duplicate class kotlin.jvm.jdk8.JvmRepeatableKt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.jvm.optionals.OptionalsKt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.random.jdk8.PlatformThreadLocalRandom found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt$asSequence$$inlined$Sequence$1 found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt$asSequence$$inlined$Sequence$2 found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt$asSequence$$inlined$Sequence$3 found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt$asSequence$$inlined$Sequence$4 found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.text.jdk8.RegexExtensionsJDK8Kt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.time.jdk8.DurationConversionsJDK8Kt found in modules jetified-kotlin-stdlib-1.9.0 (org.jetbrains.kotlin:kotlin-stdlib:1.9.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)

     Go to the documentation to learn how to <a href="d.android.com/r/tools/classpath-sync-errors">Fix dependency resolution errors</a>.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s
Running Gradle task 'assembleRelease'...                            9.1s
Gradle task assembleRelease failed with exit code 1

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>


### Screenshots or video

<details>
<summary>Screenshots or video demonstration</summary>

[Upload media here]

</details>


### Version

11.0.0

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.19.0, on macOS 14.3.1 23D60 darwin-arm64, locale
    en-EG)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] VS Code (version 1.86.2)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!

```

</details>

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -24,6 +24,8 @@ updates:
       - "mvanbeusekom"
     open-pull-requests-limit: 10
     ignore:
+      - dependency-name: "com.google.android.gms:play-services-location"
+        update-types: ["version-update:semver-major", "version-update:semver-minor", "version-update:semver-patch"]
       - dependency-name: "com.android.tools.build:gradle"
         update-types: ["version-update:semver-minor", "version-update:semver-patch"]
       - dependency-name: "junit:junit"

--- a/geolocator_android/CHANGELOG.md
+++ b/geolocator_android/CHANGELOG.md
@@ -1,3 +1,8 @@
+## 4.5.3
+
+* Adds the `notificationChannelName` field to the `ForegroundNotificationConfig` class, allowing to override the channel name.
+* Reverts `com.google.android.gms:play-services-location` to version `21.0.1`. Version `21.1.0` forces users to update AGP to version 8.0.0 and Kotlin to 1.9.0 which is not inline with stable Flutter version.
+
 ## 4.5.2
 
 * Fixes a bug where the Android `activity` is leaked when the `activiy` is detached from the Flutter application. 

--- a/geolocator_android/android/build.gradle
+++ b/geolocator_android/android/build.gradle
@@ -26,7 +26,7 @@ android {
         namespace("com.baseflow.geolocator")
     }
 
-    compileSdkVersion 33
+    compileSdk 34
 
     defaultConfig {
         minSdkVersion 16
@@ -41,8 +41,8 @@ android {
 }
 
 dependencies {
-    implementation 'com.google.android.gms:play-services-location:21.1.0'
-    implementation 'androidx.core:core:1.9.0'
+    implementation 'com.google.android.gms:play-services-location:21.0.1'
+    implementation 'androidx.core:core:1.12.0'
 
     testImplementation 'junit:junit:4.13.2'
     testImplementation 'org.mockito:mockito-core:5.1.1'

--- a/geolocator_android/android/src/main/java/com/baseflow/geolocator/GeolocatorLocationService.java
+++ b/geolocator_android/android/src/main/java/com/baseflow/geolocator/GeolocatorLocationService.java
@@ -85,8 +85,8 @@ public class GeolocatorLocationService extends Service {
   }
 
   public boolean canStopLocationService(boolean cancellationRequested) {
-    if(cancellationRequested) {
-       return listenerCount == 1;
+    if (cancellationRequested) {
+      return listenerCount == 1;
     }
     return connectedEngines == 0;
   }
@@ -143,7 +143,7 @@ public class GeolocatorLocationService extends Service {
       backgroundNotification =
           new BackgroundNotification(
               this.getApplicationContext(), CHANNEL_ID, ONGOING_NOTIFICATION_ID, options);
-      backgroundNotification.updateChannel("Background Location");
+      backgroundNotification.updateChannel(options.getNotificationChannelName());
       Notification notification = backgroundNotification.build();
       startForeground(ONGOING_NOTIFICATION_ID, notification);
       isForeground = true;
@@ -204,11 +204,19 @@ public class GeolocatorLocationService extends Service {
       WifiManager wifiManager =
           (WifiManager) getApplicationContext().getSystemService(Context.WIFI_SERVICE);
       if (wifiManager != null) {
-        wifiLock = wifiManager.createWifiLock(WifiManager.WIFI_MODE_FULL_HIGH_PERF, WIFILOCK_TAG);
+        wifiLock = wifiManager.createWifiLock(getWifiLockType(), WIFILOCK_TAG);
         wifiLock.setReferenceCounted(false);
         wifiLock.acquire();
       }
     }
+  }
+
+  @SuppressWarnings("deprecation")
+  private int getWifiLockType() {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) {
+      return WifiManager.WIFI_MODE_FULL_HIGH_PERF;
+    }
+    return WifiManager.WIFI_MODE_FULL_LOW_LATENCY;
   }
 
   class LocalBinder extends Binder {

--- a/geolocator_android/android/src/main/java/com/baseflow/geolocator/location/ForegroundNotificationOptions.java
+++ b/geolocator_android/android/src/main/java/com/baseflow/geolocator/location/ForegroundNotificationOptions.java
@@ -13,6 +13,8 @@ public class ForegroundNotificationOptions {
     @NonNull
     private final String notificationText;
     @NonNull
+    private final String notificationChannelName;
+    @NonNull
     private final AndroidIconResource notificationIcon;
     private final boolean enableWifiLock;
     private final boolean enableWakeLock;
@@ -28,6 +30,7 @@ public class ForegroundNotificationOptions {
 
     final AndroidIconResource notificationIcon = AndroidIconResource.parseArguments((Map<String, Object>)arguments.get("notificationIcon"));
     final String notificationTitle = (String) arguments.get("notificationTitle");
+    final String notificationChannelName = (String) arguments.get("notificationChannelName");
     final String notificationText = (String) arguments.get("notificationText");
     final Boolean enableWifiLock = (Boolean) arguments.get("enableWifiLock");
     final Boolean enableWakeLock = (Boolean) arguments.get("enableWakeLock");
@@ -42,6 +45,7 @@ public class ForegroundNotificationOptions {
     return new ForegroundNotificationOptions(
             notificationTitle,
             notificationText,
+            notificationChannelName,
             notificationIcon,
             enableWifiLock,
             enableWakeLock,
@@ -52,6 +56,7 @@ public class ForegroundNotificationOptions {
     private ForegroundNotificationOptions(
         @NonNull String notificationTitle,
         @NonNull String notificationText,
+        @NonNull String notificationChannelName,
         @NonNull AndroidIconResource notificationIcon,
         boolean enableWifiLock,
         boolean enableWakeLock,
@@ -59,6 +64,7 @@ public class ForegroundNotificationOptions {
         @Nullable Integer color) {
         this.notificationTitle = notificationTitle;
         this.notificationText = notificationText;
+        this.notificationChannelName = notificationChannelName;
         this.notificationIcon = notificationIcon;
         this.enableWifiLock = enableWifiLock;
         this.enableWakeLock = enableWakeLock;
@@ -74,6 +80,11 @@ public class ForegroundNotificationOptions {
     @NonNull
     public String getNotificationText() {
         return notificationText;
+    }
+
+    @NonNull
+    public String getNotificationChannelName() {
+        return notificationChannelName;
     }
 
     @NonNull

--- a/geolocator_android/example/android/app/build.gradle
+++ b/geolocator_android/example/android/app/build.gradle
@@ -1,14 +1,14 @@
+plugins {
+    id "com.android.application"
+    id "dev.flutter.flutter-gradle-plugin"
+}
+
 def localProperties = new Properties()
 def localPropertiesFile = rootProject.file('local.properties')
 if (localPropertiesFile.exists()) {
     localPropertiesFile.withReader('UTF-8') { reader ->
         localProperties.load(reader)
     }
-}
-
-def flutterRoot = localProperties.getProperty('flutter.sdk')
-if (flutterRoot == null) {
-    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
 }
 
 def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
@@ -27,12 +27,14 @@ project.getTasks().withType(JavaCompile) {
     options.compilerArgs.addAll(args)
 }
 
-apply plugin: 'com.android.application'
-apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
-
 android {
     namespace 'com.baseflow.geolocator_example'
-    compileSdkVersion 33
+    compileSdkVersion flutter.compileSdkVersion
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
 
     lintOptions {
         disable 'InvalidPackage'
@@ -41,8 +43,8 @@ android {
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "com.baseflow.geolocator_example"
-        minSdkVersion 16
-        targetSdkVersion 30
+        minSdkVersion flutter.minSdkVersion
+        targetSdkVersion flutter.targetSdkVersion
 
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName

--- a/geolocator_android/example/android/build.gradle
+++ b/geolocator_android/example/android/build.gradle
@@ -1,14 +1,3 @@
-buildscript {
-    repositories {
-        google()
-        mavenCentral()
-    }
-
-    dependencies {
-        classpath 'com.android.tools.build:gradle:7.4.2'
-    }
-}
-
 allprojects {
     repositories {
         google()

--- a/geolocator_android/example/android/settings.gradle
+++ b/geolocator_android/example/android/settings.gradle
@@ -1,15 +1,24 @@
-include ':app'
+pluginManagement {
+    def flutterSdkPath = {
+        def properties = new Properties()
+        file("local.properties").withInputStream { properties.load(it) }
+        def flutterSdkPath = properties.getProperty("flutter.sdk")
+        assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
+        return flutterSdkPath
+    }()
 
-def flutterProjectRoot = rootProject.projectDir.parentFile.toPath()
+    includeBuild("$flutterSdkPath/packages/flutter_tools/gradle")
 
-def plugins = new Properties()
-def pluginsFile = new File(flutterProjectRoot.toFile(), '.flutter-plugins')
-if (pluginsFile.exists()) {
-    pluginsFile.withReader('UTF-8') { reader -> plugins.load(reader) }
+    repositories {
+        google()
+        mavenCentral()
+        gradlePluginPortal()
+    }
 }
 
-plugins.each { name, path ->
-    def pluginDirectory = flutterProjectRoot.resolve(path).resolve('android').toFile()
-    include ":$name"
-    project(":$name").projectDir = pluginDirectory
+plugins {
+    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
+    id "com.android.application" version "7.4.2" apply false
 }
+
+include ":app"

--- a/geolocator_android/lib/src/types/foreground_settings.dart
+++ b/geolocator_android/lib/src/types/foreground_settings.dart
@@ -27,11 +27,52 @@ class AndroidResource {
 
 /// Configuration for the foreground notification. When this is provided the location service will run as a foreground service.
 class ForegroundNotificationConfig {
+  /// Creates an Android specific configuration for the [FlutterBackground] plugin.
+  ///
+  /// [notificationTitle] is the title used for the foreground service
+  /// notification.
+  /// [notificationText] is the body used for the foreground service
+  /// notification.
+  /// [notificationChannelName] is the name of the notification channel as it is
+  /// displayed in the system settings.
+  /// [notificationIcon] must be a drawable resource.
+  /// E. g. if the icon with name "background_icon" is in the "drawable"
+  /// resource folder, it should be of value
+  /// `AndroidResource(name: 'background_icon', defType: 'drawable').
+  /// [enableWifiLock] indicates wether or not a WifiLock is acquired, when the
+  /// background execution is started. This allows the application to keep the
+  /// Wi-Fi radio awake, even when the user has not used the device in a while.
+  /// [enableWakeLock] indicates wether or not a Wakelock is acquired, when the
+  /// background execution is started. If this is false then the system can
+  /// still sleep and all location events will be received at once when the
+  /// system wakes up again.
+  /// [setOngoing] indicates wether or not the displayed notification is
+  /// persistent and the user cannot dismiss it.
+  /// [color] is the accent color that is applied to the [notificationIcon].
+  const ForegroundNotificationConfig({
+    required this.notificationTitle,
+    required this.notificationText,
+    this.notificationChannelName = 'Background Location',
+    this.notificationIcon =
+        const AndroidResource(name: 'ic_launcher', defType: 'mipmap'),
+    this.enableWifiLock = false,
+    this.enableWakeLock = false,
+    this.setOngoing = false,
+    this.color,
+  });
+
   /// The title used for the foreground service notification.
   final String notificationTitle;
 
   /// The body used for the foreground service notification.
   final String notificationText;
+
+  /// The user visible name of the notification channel.
+  ///
+  /// The notification channel name will be displayed in the system settings.
+  /// The maximum recommended length is 40 characters, the name might be
+  /// truncated if it is to long. Default value: "Background Location".
+  final String notificationChannelName;
 
   /// The resource name of the icon to be used for the foreground notification.
   final AndroidResource notificationIcon;
@@ -68,37 +109,6 @@ class ForegroundNotificationConfig {
   /// If this color is null, a system default color is used.
   final Color? color;
 
-  /// Creates an Android specific configuration for the [FlutterBackground] plugin.
-  ///
-  /// [notificationTitle] is the title used for the foreground service
-  /// notification.
-  /// [notificationText] is the body used for the foreground service
-  /// notification.
-  /// [notificationIcon] must be a drawable resource.
-  /// E. g. if the icon with name "background_icon" is in the "drawable"
-  /// resource folder, it should be of value
-  /// `AndroidResource(name: 'background_icon', defType: 'drawable').
-  /// [enableWifiLock] indicates wether or not a WifiLock is acquired, when the
-  /// background execution is started. This allows the application to keep the
-  /// Wi-Fi radio awake, even when the user has not used the device in a while.
-  /// [enableWakeLock] indicates wether or not a Wakelock is acquired, when the
-  /// background execution is started. If this is false then the system can
-  /// still sleep and all location events will be received at once when the
-  /// system wakes up again.
-  /// [setOngoing] indicates wether or not the displayed notification is
-  /// persistent and the user cannot dismiss it.
-  /// [color] is the accent color that is applied to the [notificationIcon].
-  const ForegroundNotificationConfig({
-    required this.notificationTitle,
-    required this.notificationText,
-    this.notificationIcon =
-        const AndroidResource(name: 'ic_launcher', defType: 'mipmap'),
-    this.enableWifiLock = false,
-    this.enableWakeLock = false,
-    this.setOngoing = false,
-    this.color,
-  });
-
   /// Returns a JSON representation of this class.
   Map<String, dynamic> toJson() {
     return {
@@ -107,6 +117,7 @@ class ForegroundNotificationConfig {
       'notificationTitle': notificationTitle,
       'notificationIcon': notificationIcon.toJson(),
       'notificationText': notificationText,
+      'notificationChannelName': notificationChannelName,
       'setOngoing': setOngoing,
       'color': color?.value,
     };

--- a/geolocator_android/pubspec.yaml
+++ b/geolocator_android/pubspec.yaml
@@ -2,7 +2,7 @@ name: geolocator_android
 description: Geolocation plugin for Flutter. This plugin provides the Android implementation for the geolocator.
 repository: https://github.com/baseflow/flutter-geolocator/tree/main/geolocator_android
 issue_tracker: https://github.com/baseflow/flutter-geolocator/issues?q=is%3Aissue+is%3Aopen
-version: 4.5.2
+version: 4.5.3
 
 environment:
   sdk: ">=2.15.0 <4.0.0"

